### ISSUE TITLE
When the `--help` flag is passed, direct sub-commands of `kraft` print "unknown command" instead of their help

### ISSUE THREAD

**Issue by antoineco** [unikraft/kraftkit#430]
### Describe the bug

Requesting a direct sub-command's help while top-level flags are present in the arguments causes that command to print the message "unknown command" followed by a _partial_ help.

Example:

```console
$ kraft --log-type=basic version --help
unknown command "version" for "kraft version"

Usage:  kraft version

Flags:
  -h, --help   help for version
```

It doesn't occur with the top-level `kraft` command, nor with nested sub-commands. For instance, it occurs with

```console
$ kraft --log-type=basic pkg --help
                        ^
                        direct subcommand
```

but not with

```console
$ kraft --log-type=basic pkg update --help
                             ^
                             nested subcommand
```

### Expected behavior

The command's defined help is printed:

```console
$ kraft --log-type=basic version --help
Show kraft version information

USAGE
  kraft version

FLAGS
  -h, --help   help for version

INHERITED FLAGS
  --config-dir string        Path to KraftKit config directory
  --containerd-addr string   Address of containerd daemon socket
  --editor string            Set the text editor to open when prompt to edit a file
  --events-pid-file string   Events process ID used when running multiple unikernels (default "/var/kraftkit/events.pid")
  --git-protocol string      Preferred Git protocol to use (default "https")
  --http-unix-sock string    When making HTTP(S) connections, pipe requests via this shared socket
  --log-level string         Log level verbosity (default "info")
  --log-timestamps           Enable log timestamps
  --log-type string          Log type (default "fancy")
  --manifests-dir string     Path to Unikraft manifest cache
  --no-check-updates         Do not check for updates
  --no-emojis                Do not use emojis in any console output
  --no-parallel              Do not run internal tasks in parallel
  --no-prompt                Do not prompt for user interaction
  --pager string             System pager to pipe output to
  --plugins-dir string       Path to KraftKit plugin directory
  --runtime-dir string       Directory for placing runtime files (e.g. pidfiles) (default "/var/kraftkit")
  --sources-dir string       Path to Unikraft component cache
  --with-manifest strings    Paths to package or component manifests
  --with-mirror strings      Paths to mirrors of Unikraft component artifacts
```

**Comment by antoineco** [comment]
Might be fixed by #426 (and #427).

**Comment by antoineco** [comment]
The issue occurs in `cmdfactory.New`, given the following combination of `args`, while parsing the `version` cmd:

```gdb
[]string len: 3, cap: 3, [
        "--log-type=basic",
        "version",
        "--help",
]
```

https://github.com/unikraft/kraftkit/blob/8d4f1a16973643cd5591514fdf59bd118061180b/cmdfactory/builder.go#L278-L280

There are some funny checks on the position of the `--help` flag below. It is also unclear why we have a special case when the parent is the root command (which explains the behaviour specific to _direct sub-commands_):

https://github.com/unikraft/kraftkit/blob/8d4f1a16973643cd5591514fdf59bd118061180b/cmdfactory/help.go#L121-L122

The issue occurs when we enter that `if`.
With a more deeply nested command, we don't enter the `if`.
